components/ChipTextField/examples/MDCChipTextField.m

```diff
@@ -0,0 +1,360 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCChipTextField.h"
+
+#import "MDCChipTextFieldScrollView.h"
+#import "MaterialChips.h"
+
+static CGFloat const kChipsSpacing = 0.0f;
+static CGFloat const kTextToEnterPlaceholderLength = 16.0f;
+
+@interface MDCChipTextField () <MDCChipTextFieldScrollViewDataSource>
+
+@property(nonatomic, strong) MDCChipTextFieldScrollView *chipsContainerView;
+
+@property(nonatomic, readwrite, weak) NSLayoutConstraint *chipContainerViewConstraintLeading;
+@property(nonatomic, readwrite, weak) NSLayoutConstraint *chipContainerViewConstraintTrailing;
+@property(nonatomic) CGFloat chipContainerViewConstraintTrailingConstant;
```

Comment:
Why is there both chipContainerViewConstraintTrailingConstant and kChipsSpacing? Aren't they the same thing?

Reply:
chipContainerViewConstraintTrailingConstant is used for the trailing space of containerView, which is different from the spacing between chipViews.